### PR TITLE
fix: remove display of dependency problems

### PR DESCRIPTION
**Pull request by xannem**
## Summary

When compiling projects with dependencies, legend engine can return warnings from them and this can easily lead to thousands of errors being rendered on the page. These warnings are a bit extraneous and cause unnecessary clutter, so this change chooses to not display those errors.

Example of project warning panel with dependencies: 
image
 

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-application-studio/src/stores/EditorGraphState.ts [hunk 1]
```diff
@@ -186,6 +186,22 @@ export class EditorGraphState {
     return [this.error, ...this.warnings].filter(isNonNullable);
   }
 
+  removeDependencyProblems(
+    problems: Problem[] | CompilationWarning[],
+  ): Problem[] | CompilationWarning[] {
+    return problems.filter((problem) => {
+      if (problem.sourceInformation) {
```
Comment (akphi) [inline packages/legend-application-studio/src/stores/EditorGraphState.ts:193]: I don't see how this filtering work in form mode where the problem probably doesn't have any source information at all

File: packages/legend-application-studio/src/components/editor/aux-panel/Problems.tsx [hunk 2]
```diff
@@ -75,7 +75,10 @@ const ProblemItem = observer((props: { problem: Problem }) => {
 
 export const Problems = observer(() => {
   const editorStore = useEditorStore();
-  const problems = editorStore.graphState.problems;
+  const problems = editorStore.graphState.removeDependencyProblems(
```
Comment (akphi) [inline packages/legend-application-studio/src/components/editor/aux-panel/Problems.tsx:78]: I think this is unnecessary, we already handled this in `EditorGraphState`

File: packages/legend-application-studio/src/components/editor/aux-panel/AuxiliaryPanel.tsx [hunk 3]
```diff
@@ -70,7 +70,9 @@ export const AuxiliaryPanel = observer(() => {
       name: 'PROBLEMS',
       icon: undefined,
       isVisible: true,
-      counter: editorStore.graphState.problems.length,
+      counter: editorStore.graphState.removeDependencyProblems(
```
Comment (akphi) [inline packages/legend-application-studio/src/components/editor/aux-panel/AuxiliaryPanel.tsx:73]: this is redundant

File: packages/legend-application-studio/src/components/editor/StatusBar.tsx [hunk 4]
```diff
@@ -99,7 +99,9 @@ export const StatusBar = observer((props: { actionsDisabled: boolean }) => {
 
   // Problems
   const error = editorStore.graphState.error;
-  const warnings = editorStore.graphState.warnings;
+  const warnings = editorStore.graphState.removeDependencyProblems(
```
Comment (akphi) [inline packages/legend-application-studio/src/components/editor/StatusBar.tsx:102]: this is redundant, we can do this in `EditorGraphState`

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-application-studio/src/components/editor/StatusBar.tsx (hunk 4 above)
Comment (xannem) [reply packages/legend-application-studio/src/components/editor/StatusBar.tsx:102]: true!

File: packages/legend-application-studio/src/stores/EditorGraphState.ts (hunk 1 above)
Comment (xannem) [reply packages/legend-application-studio/src/stores/EditorGraphState.ts:193]: 😔

File: packages/legend-application-studio/src/stores/EditorGraphState.ts [hunk 5]
```diff
@@ -186,6 +186,22 @@ export class EditorGraphState {
     return [this.error, ...this.warnings].filter(isNonNullable);
   }
 
+  removeDependencyProblems(
```
Comment (akphi) [inline packages/legend-application-studio/src/stores/EditorGraphState.ts:189]: Let's add a comment here explaining the limitation of this approach

File: packages/legend-application-studio/src/stores/EditorGraphState.ts [hunk 6]
```diff
@@ -186,6 +186,23 @@ export class EditorGraphState {
     return [this.error, ...this.warnings].filter(isNonNullable);
   }
 
+  //TODO: remove once engine changes are added with better source information
```
Comment (akphi) [inline packages/legend-application-studio/src/stores/EditorGraphState.ts:189]:
```suggestion
/**
 * This function is temporary. There is no good way to detect if a problem not coming from
 * the main graph at the moment. In text mode, we can rely on the fact that the source information
 * has line 0 column 0. But this is not the case for form mode, so this is just temporary
 * to help with text-mode.
 */
```

**Review by akphi — APPROVED** [review]

**Pull request merged**